### ISSUE TITLE
TextInput Autocomplete prop, the typescript doesn't recognise new universal values.

### ISSUE THREAD

**Issue by luluhoc** [facebook/react-native#36441]
### Description

When using the values for autoComplete prop values from documentation I'm getting a typescript error
I have searched in the react-native package and I don't see cross-platform types in the documentation.
```
<TextInput autoComplete="given-name" />
```

Typescript error
```
No overload matches this call.
  Overload 1 of 2, '(props: TextInputProps | Readonly<TextInputProps>): TextInput', gave the following error.
    Type '"given-name"' is not assignable to type '"email" | "password" | "name" | "tel" | "off" | "birthdate-day" | "birthdate-full" | "birthdate-month" | "birthdate-year" | "cc-csc" | "cc-exp" | "cc-exp-day" | "cc-exp-month" | ... 24 more ... | undefined'.
  Overload 2 of 2, '(props: TextInputProps, context: any): TextInput', gave the following error.
    Type '"given-name"' is not assignable to type '"email" | "password" | "name" | "tel" | "off" | "birthdate-day" | "birthdate-full" | "birthdate-month" | "birthdate-year" | "cc-csc" | "cc-exp" | "cc-exp-day" | "cc-exp-month" | ... 24 more ...
```

### React Native Version

0.71.3

### Output of `npx react-native info`

```
System:
    OS: macOS 13.3
    CPU: (8) arm64 Apple M1 Pro
    Memory: 166.66 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 18.14.0 - ~/.nvm/versions/node/v18.14.0/bin/node
    Yarn: 1.22.19 - ~/Desktop/fidelity-client/node_modules/.bin/yarn
    npm: 9.3.1 - ~/.nvm/versions/node/v18.14.0/bin/npm
    Watchman: Not Found
  Managers:
    CocoaPods: 1.11.3 - /opt/homebrew/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 22.1, iOS 16.1, macOS 13.0, tvOS 16.1, watchOS 9.1
    Android SDK: Not Found
  IDEs:
    Android Studio: 2021.3 AI-213.7172.25.2113.9014738
    Xcode: 14.1/14B47b - /usr/bin/xcodebuild
  Languages:
    Java: 19.0.1 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 18.2.0 => 18.2.0 
    react-native: 0.71.3 => 0.71.3 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```

### Steps to reproduce

Types from react-native package for autocomplete.
```
autoComplete?:
    | 'birthdate-day'
    | 'birthdate-full'
    | 'birthdate-month'
    | 'birthdate-year'
    | 'cc-csc'
    | 'cc-exp'
    | 'cc-exp-day'
    | 'cc-exp-month'
    | 'cc-exp-year'
    | 'cc-number'
    | 'email'
    | 'gender'
    | 'name'
    | 'name-family'
    | 'name-given'
    | 'name-middle'
    | 'name-middle-initial'
    | 'name-prefix'
    | 'name-suffix'
    | 'password'
    | 'password-new'
    | 'postal-address'
    | 'postal-address-country'
    | 'postal-address-extended'
    | 'postal-address-extended-postal-code'
    | 'postal-address-locality'
    | 'postal-address-region'
    | 'postal-code'
    | 'street-address'
    | 'sms-otp'
    | 'tel'
    | 'tel-country-code'
    | 'tel-national'
    | 'tel-device'
    | 'username'
    | 'username-new'
    | 'off'
    | undefined;
```

Types in documentation for universal that are not recognised
```
The following values work across platforms:

additional-name
address-line1
address-line2
cc-number
country
current-password
email
family-name
given-name
honorific-prefix
honorific-suffix
name
new-password
off
one-time-code
postal-code
street-address
tel
username
```

### Snack, code example, screenshot, or link to a repository

no snack

**Comment by luluhoc** [comment]
By the way the same happens with 0.71.4

**Comment by rahman** [comment]
still an issue in 0.71.7

**Comment by lunaleaps** [comment]
Can you clarify what you mean by "Types in documentation for universal that are not recognised"?

**Comment by Bi0max** [comment]
@lunaleaps , in the docs there is a list of values, which work across platforms: https://reactnative.dev/docs/textinput#autocomplete

But some of them are not recognized by Typescript, like "given-name" or "family-name".

**Comment by lunaleaps** [comment]
@Bi0max This has been fixed here: https://github.com/facebook/react-native/commit/94356e14ec0562a1fd5a208d93021f102ba9565e and will be shipped in 0.73. I can list it as a pick for 0.72 here: https://github.com/reactwg/react-native-releases/discussions/54
